server/src/main/java/org/opensearch/search/fetch/ShardFetchRequest.java

```diff
@@ -67,9 +67,9 @@ public class ShardFetchRequest extends TransportRequest {
 
     private ScoreDoc lastEmittedDoc;
 
-    public ShardFetchRequest(ShardSearchContextId contextId, IntArrayList list, ScoreDoc lastEmittedDoc) {
+    public ShardFetchRequest(ShardSearchContextId contextId, List<Integer> list, ScoreDoc lastEmittedDoc) {
```

Comment:
Nitpick, but this can be a `Collection<Integer>` if you don't need any properties specific to a List.